diff --git a/src/core-setup/cross/armel/toolchain.cmake b/src/core-setup/cross/armel/toolchain.cmake
index 8efb6f32..65cb1a57 100644
--- a/src/core-setup/cross/armel/toolchain.cmake
+++ b/src/core-setup/cross/armel/toolchain.cmake
@@ -5,12 +5,14 @@ set(CMAKE_SYSTEM_VERSION 1)
 set(CMAKE_SYSTEM_PROCESSOR armv7l)
 
 ## Specify the toolchain
-set(TOOLCHAIN "arm-linux-gnueabi")
+set(TOOLCHAIN "arm-nilrt-linux-gnueabi")
 set(TOOLCHAIN_PREFIX ${TOOLCHAIN}-)
 
-add_compile_options(-target armv7-linux-gnueabi)
+add_compile_options(-target ${TOOLCHAIN})
 add_compile_options(-mthumb)
 add_compile_options(-mfpu=vfpv3)
+add_compile_options(-mfloat-abi=softfp)
+add_compile_options(-march=armv7l)
 add_compile_options(--sysroot=${CROSS_ROOTFS})
 
 set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -target ${TOOLCHAIN}")
@@ -25,8 +27,7 @@ if("$ENV{__CrossToolChainTargetRID}" MATCHES "tizen.*")
     set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -B${CROSS_ROOTFS}/usr/lib/gcc/${TIZEN_TOOLCHAIN}")
     set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -L${CROSS_ROOTFS}/usr/lib/gcc/${TIZEN_TOOLCHAIN}")
 else()
-    set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -B${CROSS_ROOTFS}/usr/lib/gcc/${TOOLCHAIN}")
-    set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -L${CROSS_ROOTFS}/usr/lib/gcc/${TOOLCHAIN}")
+    set(CROSS_LINK_FLAGS "${CROSS_LINK_FLAGS} -mthumb -mfpu=vfpv3 -mfloat-abi=softfp -march=armv7l")
 endif()
 
 set(CMAKE_EXE_LINKER_FLAGS    "${CMAKE_EXE_LINKER_FLAGS}    ${CROSS_LINK_FLAGS}" CACHE STRING "" FORCE)
diff --git a/src/core-setup/src/corehost/build.sh b/src/core-setup/src/corehost/build.sh
index 95ef76dc..5a1616c1 100755
--- a/src/core-setup/src/corehost/build.sh
+++ b/src/core-setup/src/corehost/build.sh
@@ -180,8 +180,8 @@ if [ "$__CrossBuild" == 1 ]; then
 fi
 
 # __rid_plat is the base RID that corehost is shipped for, effectively, the name of the folder in "runtimes/{__rid_plat}/native/" inside the nupkgs.
-__rid_plat=
-init_rid_plat
+__rid_plat=linux
+#init_rid_plat
 
 if [ -z $__rid_plat ]; then
     echo "Unknown base rid (eg.: osx.10.12, ubuntu.14.04) being targeted"
@@ -231,6 +231,9 @@ if [ $__CrossBuild == 1 ]; then
     if command -v "clang-3.9" > /dev/null 2>&1; then
         export CC="$(command -v clang-3.9)"
         export CXX="$(command -v clang++-3.9)"
+    elif command -v clang > /dev/null 2>&1; then
+        export CC="$(command -v clang)"
+        export CXX="$(command -v clang++)"
     else
         echo "Unable to find Clang 3.9 Compiler"
         echo "Install clang-3.9 for cross compilation"
