diff --git a/cross/armel/toolchain.cmake b/cross/armel/toolchain.cmake
index 86e2306..e26d844 100644
--- a/cross/armel/toolchain.cmake
+++ b/cross/armel/toolchain.cmake
@@ -6,7 +6,7 @@ set(CMAKE_SYSTEM_PROCESSOR armv7l)
 
 set(TOOLCHAIN "arm-linux-gnueabi")
 
-add_compile_options(-target armv7-linux-gnueabi)
+add_compile_options(-target ${TOOLCHAIN})
 add_compile_options(-mthumb)
 add_compile_options(-mfpu=vfpv3)
 add_compile_options(-mfloat-abi=softfp)
diff --git a/dir.props b/dir.props
index db5ffe1..ee64642 100644
--- a/dir.props
+++ b/dir.props
@@ -104,7 +104,7 @@
 
   <PropertyGroup>
      <ArmEnvironmentVariables Condition="'$(ArmEnvironmentVariables)' == ''">ROOTFS_DIR=$(BaseIntermediatePath)crossrootfs/arm</ArmEnvironmentVariables>
-     <ArmEnvironmentVariables Condition="'$(Platform)' == 'armel'">ROOTFS_DIR=$(BaseIntermediatePath)crossrootfs/armel</ArmEnvironmentVariables>
+     <!--<ArmEnvironmentVariables Condition="'$(Platform)' == 'armel'">ROOTFS_DIR=$(BaseIntermediatePath)crossrootfs/armel</ArmEnvironmentVariables>-->
    </PropertyGroup>
 
   <Import Project="$(TargetInfoProps)" Condition="$(GeneratingStaticPropertiesFile) != 'true' AND Exists('$(TargetInfoProps)')" />
diff --git a/repos/coreclr.proj b/repos/coreclr.proj
index 3d76a79..e8b1726 100644
--- a/repos/coreclr.proj
+++ b/repos/coreclr.proj
@@ -6,7 +6,7 @@
     <BuildArguments Condition="'$(SkipDisablePgo)' != 'true'">$(BuildArguments) -nopgooptimize</BuildArguments>
     <BuildArguments Condition="'$(OS)' != 'Windows_NT'">$(BuildArguments) msbuildonunsupportedplatform</BuildArguments>
     <BuildArguments Condition="'$(UseSystemLibraries)' == 'true' AND '$(OS)' != 'Windows_NT'">$(BuildArguments) cmakeargs -DCLR_CMAKE_USE_SYSTEM_LIBUNWIND=TRUE</BuildArguments>
-    <BuildArguments Condition="$(Platform.Contains('arm'))">$(BuildArguments) skipnuget cross -skiprestore cmakeargs -DFEATURE_GDBJIT=TRUE</BuildArguments>
+    <BuildArguments Condition="$(Platform.Contains('arm'))">$(BuildArguments) cross skipnuget -skiprestore cmakeargs -DFEATURE_GDBJIT=TRUE</BuildArguments>
     <!-- Portable builds only apply to Linux - Mac and Windows are both always portable.
          Additionally, Linux builds are portable by default and only have a switch to turn it off -->
     <BuildArguments Condition="'$(TargetOS)' == 'Linux' and '$(PortableBuild)' == 'false'">$(BuildArguments) -PortableBuild=false</BuildArguments>
diff --git a/repos/corefx.proj b/repos/corefx.proj
index b09fb31..fafd93f 100644
--- a/repos/corefx.proj
+++ b/repos/corefx.proj
@@ -13,6 +13,7 @@
     <BuildArguments>$(BuildArguments) -buildArch=$(Platform)</BuildArguments>
     <BuildArguments>$(BuildArguments) -portable=$(OverridePortableBuild) </BuildArguments>
     <BuildArguments>$(BuildArguments) -BuildTests=false</BuildArguments>
+    <BuildArguments>$(BuildArguments) /p:CoreCLROverridePath=$(MSBuildThisFileDirectory)/../coreclr/bin/Produce/Linux.armel.Release</BuildArguments>
     <BuildArguments>$(BuildArguments) -PackageRid=$(OverrideTargetRid)</BuildArguments>
     <BuildArguments>$(BuildArguments) --</BuildArguments>
     <BuildArguments>$(BuildArguments) /p:BuildAllPackages=true</BuildArguments>
